src/bp/ui-studio/src/web/views/FlowBuilder/diagram/index.tsx

```diff
@@ -63,17 +66,31 @@ export default class FlowBuilder extends Component<Props> {
 
   componentDidMount() {
     this.props.fetchFlows()
-    ReactDOM.findDOMNode(this.diagramWidget).addEventListener('click', this.onDiagramClick)
-    ReactDOM.findDOMNode(this.diagramWidget).addEventListener('dblclick', this.onDiagramDoubleClick)
+    ReactDOM.findDOMNode(this.diagramWidget).addEventListener('mousedown', this.onDiagramMouseDown)
+    ReactDOM.findDOMNode(this.diagramWidget).addEventListener('mouseup', this.onDiagramMouseUp)
+    ReactDOM.findDOMNode(this.diagramWidget).addEventListener('mousemove', this.onDiagramMouseMove)
     document.getElementById('diagramContainer').addEventListener('keydown', this.onKeyDown)
   }
 
   componentWillUnmount() {
-    ReactDOM.findDOMNode(this.diagramWidget).removeEventListener('click', this.onDiagramClick)
-    ReactDOM.findDOMNode(this.diagramWidget).removeEventListener('dblclick', this.onDiagramDoubleClick)
+    ReactDOM.findDOMNode(this.diagramWidget).removeEventListener('mousedown', this.onDiagramMouseDown)
+    ReactDOM.findDOMNode(this.diagramWidget).removeEventListener('mouseup', this.onDiagramMouseUp)
+    ReactDOM.findDOMNode(this.diagramWidget).removeEventListener('mousemove', this.onDiagramMouseMove)
     document.getElementById('diagramContainer').removeEventListener('keydown', this.onKeyDown)
   }
 
+  onDiagramMouseMove = () => {
+    this.moved = true
+  }
+
+  onDiagramMouseDown = () => {
+    this.moved = false
+  }
+
+  onDiagramMouseUp = () => {
+    this.onDiagramClick()
```

Comment:
Not sure the alias is required, you could just set onDiagramClick in your mouseup event listener.

The naming of this.moved doesn't really reflect its purpose, I suggest using this.movingNode